From 2d3273f96f7527690b3d2e16d1aa766a8e19c376 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@hoyer.xyz>
Date: Mon, 18 Dec 2017 10:01:33 +0100
Subject: [PATCH] Merge pull request #325 from danimo/dasd_cio_free

95dasd_mod: do not set module parameters if dasd_cio_free is not present
---
 modules.d/95dasd_mod/parse-dasd-mod.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/95dasd_mod/parse-dasd-mod.sh b/modules.d/95dasd_mod/parse-dasd-mod.sh
index 92ee1a79..b4c01a32 100755
--- a/modules.d/95dasd_mod/parse-dasd-mod.sh
+++ b/modules.d/95dasd_mod/parse-dasd-mod.sh
@@ -7,7 +7,7 @@ done
 
 mod_args="${mod_args#*,}"
 
-if [ -n "$mod_args" ]; then
+if [ -x /sbin/dasd_cio_free -a -n "$mod_args" ]; then
     [ -d /etc/modprobe.d ] || mkdir -m 0755 -p /etc/modprobe.d
     echo "options dasd_mod dasd=$mod_args" >> /etc/modprobe.d/dasd_mod.conf
 fi

